Automattic/jetpack
# Move Photon to a separate package - "Image CDN"

## Proposed changes:

* Add Image CDN package;
* Add Image CDN as a separate optimization module to Jetpack Boost.
* Updated Boost compatibility file to sync status between image-cdn in boost and photon in jetpack.
* Disable photon in jetpack when image-cdn is present and use that instead

### Other information:


## Jetpack product discussion



No.

## Does this pull request change what data or activity we track or use?



No.

## Testing instructions:






* Activate Image CDN and test if image URLs start with https://i0.wp.com
* Make sure **Jetpack -> Settings -> Performance -> "Speed up image load times"** syncs with **Boost -> "Image CDN"** when you toggle the status on either place
* Make sure, photon inside Jetpack does not load if Boost has Image CDN enabled. There is no visible way to verify this as the functionality is almost same between Image CDN and Photon, But, you can try hacking by adding `wp_die()` or something in the photon file in Jetpack.

> **Note**
> I have investigated the failing e2e tests. The tests seem broken because Jetpack asks to prove humanity when the e2e tests are running. The tests should be updated and in a different PR.
